### DISCUSSION TITLE
Hours worked

### DISCUSSION THREAD

**Discussion by IllustratedMan-code** [IllustratedMan-code/SeniorProject#11] [General]
# List your hours!

List the individual hours worked in this discussion. 

# Group hours

Group hours correspond to meetings and any extra time we spend working together "in person".

9/4/2023: 2 hours
9/11/2023: 2 hours
9/18/2023: 2 hours
9/25/2023: 2 hours
10/2/2023: 1 hour
10/9/2023: 1 hour
10/16/2023: 1 hour
10/23/2023: 1 hour
10/30/2023: 1 hour
11/6/2023: 1 hour
11/20/2023: 1 hour

12/9/2023: hackathon

1/9/2024: 1 hour
2/5/2024: 1 hour
3/5/2024: 1 hour
3/21/2024: 1 hour

15 total group hours as of 11/23/2023

**Comment by IllustratedMan-code** [comment]
# David
| hours worked | date | What I did |
|----------------|-------|------------|
| ~10 | pre 10/2/2023 (can't remember specific dates) | Brainstormed game, setup Github, did design assignments, prototyped mechanics, created assets |
| 3 | 10/2/2023 |  Researched accessibility features of Godot, did design assignment (module 7) |
| 10 |  10/9-10/15/2023 | Created assets and player prototype|
| 10 | 10/16-10/22/2023 | Designed and implemented a mutation minigame |
| 5 | 10/23-10/29/2023 | Responded to peer review and implemented changes to mutation minigame |
| 12 | 11/6-11/12/2023 | Created bacteriophage and spider [assets](https://github.com/quaternary-game/quaternary/issues/17) | 
| 4 | 11/13-11/20/2023 | Created bacteria asset |

As of 11/24/2023: 51 individual hours + 15 group hours = 66 total hours

**Reply by IllustratedMan-code** [reply]
| hours worked | date | What I did |
|----------------|-------|------------|
|~40 | pre 3/11/24 | General updates/bugfixes, discussions/brainstorms |
|40 | 3/11/24-3/15/24 | Main game UI |
| 40 | 3/25/24-3/29/24| Working towards level 1 on Main game (trait creation/UI development) |

**Comment by IllustratedMan-code** [comment]
# Alex
| hours worked | date | What I did |
|----------------|-------|------------|
| ~8 | Earlier than 10/12 | Brainstorming, class assignments, Godot tutorial and reading docs |
| 9 | 10/12 - 10/16 | working on mini game POC, implementing basic functionality of a TD game |
| 4 | 10/20-10/23 | writing slides and recording presentation, creating general sound design |
| 1 | 10/26 | peer project reviews |
| 1 | 10/30 | brainstorm mini game details | 
| 4 | 11/5 | started working on punnett square game POC |
| 4 | 11/6-11/7 | working on punnett square game POC |
| 2 | 11/12 | finished up punnett square game |
| 2 | 11/19 | compiling final report, adding docstrings to code |
| 4 | 1/14-1/15 | working on pausing and scene switching |
| 3 | 1/15 - 1/22 | learning gut testing framework, writing tests |
| 3 | 2/6 | fixed bugs, worked on loading screens |
| 2 | 2/11 | designed slide deck for expo presentation |
| 3 | 2/12 | started implementing game saves, debugging loading screens |
| 2 | 3/21 | General bug fixes and improvements to punnett square mini game |
| 3 | 3/25 | Improving upon pause function and menu | 
| 3 | 3/26 | improving upon mutation master, added features and fixed bugs |
| 3 | 3/27 | started sound design system |
| 5 | 3/29 | working on sound design system, added blur background to pause menu |
| 2 | 3/30 | finished up sound design |
| 5 | 4/4-4/5 | added settings menu |
| 1 | 4/7 | punnett square tutorial |
| 5 | 4/8 | wrap up changes for expo (color blind filters, end game screens, punnett sq changes) |


As of 4/14/2024: 35 individual hrs + 15 group hrs (Fall) + 44 individual + 9 group (spring) = 103 hrs

**Comment by IllustratedMan-code** [comment]
# Connor
| hours worked | date | What I did |
|----------------|-------|------------|
| ~13 | pre 10/12 |Brainstorming, class assignments, Godot tutorial and reading docs |
| 7 | 10/12 - 10/16 | worked on proof of concept body invaders minigame
| 2 | 10/22 | Wrote and recorded presentation
| 3 | 10/23 | started work on codon building game
| 1 | 10/28 | Completed project peer reviews
| 6 | 11/6 | Completed rough draft for codon building game
| 6 | 11/20 | Completed final version of codon building game, merged with developer branch
| 4 | 12/9 | Updated project to Godot 4.2, working session/various fixes
| 4 | 1/9 | Created main menu, fixed issues with codon minigame
| 3 | 1/16 | Worked on the main menu, created and integrated minigame submenu
| 2 | 2/7 | Attempted to fix flickering in codon minigame, updated pause menu for codon game
| 6 | 2/13 | Added consistent UI theming for all menus/games
| 10 | 3/26 | Created Quaternary poster
| 1 | 4/1 | Fixed flickering in codon minigame
| 3 | 4/4 | Started work on tutorial
| 8 | 4/5 | Finished early tutorial tool build
| 5 | 4/5 | Working session/codon tutorial
| 3 | 4/8 | Expo prep

As of 04/14: 87 individual hours + 15 group hours (Fall) + 8 group hours (Spring) = 110 hours

**Comment by IllustratedMan-code** [comment]
# John
| hours worked | date | What I did |
|----------------|-------|------------|
| ~20 | Before 10/12 | Brainstorming, Class assignments, Godot Tutorial/Docs Reading, Godot Standards/Internationalization Research, Defined Contributing Guidelines, and Setup Project/Repository |
| 5 | 10/12 | PR Code Review, Godot ECS Research, GDExtension research |
| 2 | 10/22 | Created Presentation Slides + Recording |
| 3 | 10/23 | Godot CI/CD Research, Setup Godot Export Presets |
| 1 | 10/26 | Performed project peer reviews |
| 2 | 11/7 | Entity Stress Test Setup |
| 1 | 11/20 | PR Code Review |

Total for first semester: 34 individual hours + 15 group hours = 49 hours

**Reply by john-whiting** [reply]
| hours worked | date | What I did |
| -------------- | ---- | ----------- |
| 4   | 01/20 | Added rudimentary entities with trait system and stress test |
| 2   | 02/06 | Added herbivore-like properties to entity |
| 4   | 02/22 | Refactored entity properties, added calorie trait, and added photoautotroph trait |
| 13 | 04/02 | Genotype refactor into separate dataclasses with research and Godot editor support |
| 2   | 04/03 | Added genotype cloning and merging |
| 6   | 04/04 | Added asexual and sexual traits for realistic entity reproduction |
| 7   | 04/07 | Remade tutorial system to add more capabilities and added a level 1 tutorial |
| 3   | spring | Pull request reviews |
| 4   | spring | Miscellaneous bug fixes |

Spring semester individual hours: 45 hours
Spring semester meetings: 6 hours

---

Total Spring Semester Hours: 51 hours

**Reply by john-whiting** [reply]
Total for both semesters: 49 hours + 51 hours = 100 hours
